tests/end_to_end/models/participants.py

```diff
@@ -0,0 +1,431 @@
+            log.error(f"Failed to sign the CSR: {e}")
+            raise e
+        return True
+
+    def modify_plan(self, new_rounds=None, num_collaborators=None):
+        """
+        Modify the plan to train the model
+        Args:
+            new_rounds (int): Number of rounds to train
+            num_collaborators (int): Number of collaborators
+        Returns:
+            bool: True if successful, else False
+        """
+        self.plan_path = os.path.join(self.workspace_path, "plan", "plan.yaml")
+        log.info(f"Modifying the plan at {self.plan_path}")
+        # Open the file and modify the entries
+        self.rounds_to_train = new_rounds if new_rounds else self.rounds_to_train
+        self.num_collaborators = num_collaborators if num_collaborators else self.num_collaborators
+
+        with open(self.plan_path) as fp:
+            data = yaml.load(fp, Loader=yaml.FullLoader)
+
+        data["aggregator"]["settings"]["rounds_to_train"] = int(self.rounds_to_train)
+        data["data_loader"]["settings"]["collaborator_count"] = int(self.num_collaborators)
+
+        with open(self.plan_path, "w+") as write_file:
+            yaml.dump(data, write_file)
+
+        log.info(
+            f"Modified the plan to train the model for collaborators {self.num_collaborators} and {self.rounds_to_train} rounds"
+        )
+        return True
+
+    def initialize_plan(self, agg_domain_name):
+        """
+        Initialize the plan
+        Args:
+            agg_domain_name (str): Aggregator domain name
+        Returns:
+            bool: True if successful, else False
+        """
+        try:
+            log.info("Initializing the plan. It will take some time to complete..")
+            return_code, _, error = sh.run_command(f"fx plan initialize -a {agg_domain_name}", work_dir=self.workspace_path)
+            if return_code != 0:
+                log.error(f"Failed to initialize the plan: {error}")
+                raise Exception(f"Failed to initialize the plan: {error}")
+
+            log.info(f"Initialized the plan for the workspace {self.workspace_name}")
+        except Exception as e:
+            log.error(f"Failed to initialize the plan: {e}")
+            raise e
+        return True
+
+    def certify_workspace(self):
+        """
+        Certify the workspace
+        Returns:
+            bool: True if successful, else False
+        """
+        try:
+            return_code, _, error = sh.run_command("fx workspace certify", work_dir=self.workspace_path)
+            if return_code != 0:
+                log.error(f"Failed to certify the workspace: {error}")
+                raise Exception(f"Failed to certify the workspace: {error}")
+
+            log.info(f"Certified the workspace {self.workspace_name}")
+        except Exception as e:
+            log.error(f"Failed to certify the workspace: {e}")
+            raise e
+        return True
+
+    def certify_agg_request(self, agg_domain_name):
```

Comment:
For the sake of symmetry, let's rename this to simply `certify_aggregator`.

Reply:
This is done